app/Http/Controllers/WikiEntityImportController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\WikiEntityImportStatus;
+use Illuminate\Http\Request;
+use App\Wiki;
+use App\WikiEntityImport;
+use App\Jobs\WikiEntityImportDummyJob;
+use Carbon\Carbon;
+
+class WikiEntityImportController extends Controller
+{
+    public function get(Request $request): \Illuminate\Http\JsonResponse
+    {
+        $wiki = Wiki::find($request->input('wiki'));
+        if (!$wiki) {
+            abort(404, 'No such wiki');
+        }
+
+        $imports = $wiki->wikiEntityImports()->get();
+        return response()->json(['data' => $imports]);
+    }
+
+    public function create(Request $request): \Illuminate\Http\JsonResponse
+    {
+        $wiki = Wiki::find($request->input('wiki'));
+        if (!$wiki) {
+            abort(404, 'No such wiki');
+        }
+
+        $imports = $wiki->wikiEntityImports()->get();
+        foreach ($imports as $import) {
+            if ($import->status === WikiEntityImportStatus::Success) {
+                return response()
+                    ->json(['error' => 'Wiki "'.$wiki->domain.'" already has performed a successful entity import'])
+                    ->setStatusCode(400);
+            }
+            if ($import->status === WikiEntityImportStatus::Pending) {
+                return response()
+                    ->json(['error' => 'Wiki "'.$wiki->domain.'" currently has a pending entity import'])
+                    ->setStatusCode(400);
+            }
+        }
+
+        $import = $wiki->wikiEntityImports()->create([
+            'status' => WikiEntityImportStatus::Pending,
+            'started_at' => Carbon::now(),
+            'payload' => $request->all(),
+        ]);
+
+        dispatch(new WikiEntityImportDummyJob(
+            wikiId: $wiki->id,
+            sourceWikiUrl: $request->input('source_wiki_url'),
+            importId: $import->id,
+            entityIds: $request->input('entity_ids'),
+        ));
+
+        return response()->json(['data' => $import]);
+    }
+
+    public function update(Request $request): \Illuminate\Http\JsonResponse
```

Comment:
is the idea of this method that we have a mechanism to stop an import?

Reply:
The idea is that the job can signal back the outcome of its work. I.e. the user POSTs an update via UI, a job is created, and then the job can use the backend route to tell API whether the run succeeded or failed.

Reply:
ah right, got it, thanks